ledger/acctonline.go

```diff
@@ -504,13 +504,32 @@ func (ao *onlineAccounts) postCommit(ctx context.Context, dcc *deferredCommitCon
 func (ao *onlineAccounts) postCommitUnlocked(ctx context.Context, dcc *deferredCommitContext) {
 }
 
-// OnlineTotals return the total online balance for the given round.
-func (ao *onlineAccounts) OnlineTotals(rnd basics.Round) (basics.MicroAlgos, error) {
+// onlineTotals return the total online balance for the given round.
+func (ao *onlineAccounts) onlineTotals(rnd basics.Round) (basics.MicroAlgos, error) {
 	ao.accountsMu.RLock()
 	defer ao.accountsMu.RUnlock()
 	return ao.onlineTotalsImpl(rnd)
 }
 
+// onlineTotalsEx return the total online balance for the given round for extended rounds range
+// by looking into DB
+func (ao *onlineAccounts) onlineTotalsEx(rnd basics.Round) (basics.MicroAlgos, error) {
+	ao.accountsMu.RLock()
+	totalsOnline, err := ao.onlineTotalsImpl(rnd)
+	ao.accountsMu.RUnlock()
+	if err == nil {
+		return totalsOnline, err
+	}
+
+	var roundOffsetError *RoundOffsetError
+	if !errors.As(err, &roundOffsetError) {
+		ao.log.Errorf("onlineTotalsImpl error:  %w", err)
```

Comment:
So you are logging this at error level because it should never happen, but we fail over to the DB lookup anyway just in case it does? Error level is reported to telemetry so I assume you want to make sure this is never used, is that right?

Reply:
you are right.

Reply:
OK SGTM, we should probably make a list of this and any other error/warning log messages your team would want to be notified if they occurred .. we can set up monitoring for patterns